db/db_impl_debug.cc

```diff
@@ -126,7 +126,7 @@ Status DBImpl::TEST_WaitForCompact() {
 
   InstrumentedMutexLock l(&mutex_);
   while ((bg_bottom_compaction_scheduled_ || bg_compaction_scheduled_ ||
-          bg_flush_scheduled_) &&
+          bg_flush_scheduled_ || (waitUnscheduled && unscheduled_compactions_)) &&
```

Comment:
the camel-case is more for function names. variable names can be underscore-separated lowercase (we follow google style guide if you're interested)